org.eclipse.tm4e.ui/src/main/java/org/eclipse/tm4e/ui/themes/Theme.java

```diff
@@ -136,23 +150,9 @@ public boolean isDefault() {
 
 	@Override
 	public void initializeViewerColors(StyledText styledText) {
-		ITokenProvider tokenProvider = getTokenProvider();
-		Color background = tokenProvider.getEditorBackground();
-		if (background != null) {
-			styledText.setBackground(background);
-		}
-		Color foreground = tokenProvider.getEditorForeground();
-		if (foreground != null) {
-			styledText.setForeground(foreground);
-		}
-		Color selectionBackground = tokenProvider.getEditorSelectionBackground();
-		if (selectionBackground != null) {
-			styledText.setSelectionBackground(selectionBackground);
-		}
-		Color selectionForeground = tokenProvider.getEditorSelectionForeground();
-		styledText.setSelectionForeground(null);
-		if (selectionForeground != null) {
-			styledText.setSelectionForeground(selectionForeground);
-		}
+		styledText.setBackground(getEditorBackground());

```

Comment:
Couldn't `getEditorBackground()` return null if there is no need to override it?
I don't think TM4E should call setBackground at all if it has no good reason to do it (and same for other colors)

Reply:
hm, getters can return null. I see it in that way :  userDefined >> themeColor >> null. If you have another vision, please, tell me.

Reply:
Yes, but if StyledTest already has defined color that come from someplace else, and that userDefined == null && themeColor == null, then the default color pre-set in the StyledTest should be kept. Here it would remove it.

Reply:
Ok, tomorrow I'll push all changes.